cdap-support-bundle/src/main/java/io/cdap/cdap/handlers/HealthCheckHttpHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.handlers;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.implementation.HealthCheckImplementation;
+import io.cdap.cdap.proto.element.EntityType;
+import io.cdap.cdap.proto.id.InstanceId;
+import io.cdap.cdap.proto.security.StandardPermission;
+import io.cdap.cdap.security.spi.authorization.ContextAccessEnforcer;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+/**
+ * Health Check HTTP Handler.
+ */
+@ApplicationScoped
+@Path(Constants.Gateway.API_VERSION_3)
+public class HealthCheckHttpHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HealthCheckHttpHandler.class);
+  private static final Gson GSON = new Gson();
+  private final HealthCheckImplementation healthCheckImplementation;
+  private final ContextAccessEnforcer contextAccessEnforcer;
+
+  @Inject
+  HealthCheckHttpHandler(HealthCheckImplementation healthCheckImplementation, ContextAccessEnforcer contextAccessEnforcer) {
+    this.healthCheckImplementation = healthCheckImplementation;
+    this.contextAccessEnforcer = contextAccessEnforcer;
+  }
+
+  @GET
+  @Path("/health")
+  public void call(HttpRequest request, HttpResponder responder) {
+    /** ensure the user has authentication to get health check\ */
+    contextAccessEnforcer.enforceOnParent(EntityType.HEALTH_CHECK, InstanceId.SELF, StandardPermission.GET);
```

Comment:
I think it's fine here to just do something like:
```
contextAccessEnforcer.enforce(InstanceId.SELF, InstancePermission.HEALTH_CHECK);
```
where InstancePermission.HEALTH_CHECK is a new permission on the InstanceId.